src/go/cmd/pkg.go

```diff
@@ -294,6 +298,97 @@ func (pkg Pkg) addTypeNavigator(oriVal string, imports map[string]string) string
 	}
 }
 
+// TypeAlias represents a type exported from one package but defined in another. In code
+// this looks like "type Event = log.Event".
+type TypeAlias struct {
+	// Name in the exporting package e.g. "Event"
+	Name string
+	// Package containing the alias
+	Package *Pkg
+	// QualifiedName of the source type e.g. "github.com/Azure/azure-sdk-for-go/sdk/internal/log.Event"
+	QualifiedName string
+	// SourceModPath is the path of the module defining the type e.g. "github.com/Azure/azure-sdk-for-go/sdk/internal"
+	SourceModPath string
+
+	// resolved indicates whether the alias has been resolved
+	resolved bool
+}
+
+// Resolve adds review content for the alias. If def is nonzero i.e., it carries a syntax node for the type definition,
+// Resolve adds that definition to the package exporting the alias. Otherwise, Resolve adds a SimpleType representing the
+// alias to the review.
+func (a *TypeAlias) Resolve(def typeDef) error {
+	if a.resolved {
+		// this should never happen but if it does, it's a bug we want to know about
+		return fmt.Errorf("alias %s already resolved", a.Name)
+	}
+	if def != (typeDef{}) {
+		a.Package.types[a.Name] = def
+	}
+	level := DiagnosticLevelInfo
+	originalName := a.QualifiedName
+	if _, after, found := strings.Cut(a.QualifiedName, a.Package.modulePath); found {
+		originalName = strings.TrimPrefix(after, "/")
+	} else {
+		level = DiagnosticLevelWarning
+	}
+	var t TokenMaker
+	if def.n == nil || def.p == nil {
+		t = a.Package.c.addSimpleType(*a.Package, a.Name, a.Package.Name(), a.QualifiedName, nil)
+	} else {
+		switch n := def.n.Type.(type) {
+		case *ast.InterfaceType:
+			t = a.Package.c.addInterface(*def.p, a.Name, a.Package.Name(), n, nil)
+		case *ast.StructType:
+			t = a.Package.c.addStruct(*def.p, a.Name, a.Package.Name(), def.n, nil)
+			hoistMethodsForType(def.p, a.Name, a.Package)
+			// ensure that all struct field types that are structs are also aliased from this package
+			for _, field := range n.Fields.List {
+				fieldTypeName := unwrapStructFieldTypeName(field)
+				if fieldTypeName == "" {
+					// we can ignore this field
+					continue
+				}
+
+				// ensure that our package exports this type
+				found := false
+				for _, ta := range a.Package.TypeAliases {
+					if ta.Name == fieldTypeName {
+						found = true
+						break
+					}
+				}
+				if found {
+					continue
+				}
+
+				// no alias, add a diagnostic
+				a.Package.diagnostics = append(a.Package.diagnostics, Diagnostic{
+					Level:    DiagnosticLevelError,
+					TargetID: t.ID(),
+					Text:     missingAliasFor + fieldTypeName,
+				})
+			}
+		case *ast.Ident:
+			t = a.Package.c.addSimpleType(*a.Package, a.Name, a.Package.Name(), def.n.Type.(*ast.Ident).Name, nil)
+			hoistMethodsForType(def.p, a.Name, a.Package)
+		default:
+			fmt.Printf("unexpected node type %T\n", def.n.Type)
```

Comment:
ErrorF

Reply:
This is another chunk I moved without changing. We have this case because the switch isn't exhaustive. The way it works now is when we resolve a type alias to a definition using syntax we don't explicitly handle, we give up and add the alias to the API review as-is. Returning an error instead would make the module unreviewable. Not necessarily a bad idea but we should add at least the case we need today before making the change. I opened #8670 to track that.